### PR TITLE
ISPN-9485 RoundRobinBalancingStrategy always starts from server 0

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-9485

### CODE REVIEW

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/RoundRobinBalancingStrategy.java [hunk 1]
```diff
@@ -28,7 +29,7 @@ public void setServers(Collection<SocketAddress> servers) {
       this.servers = servers.toArray(new SocketAddress[servers.size()]);
       // keep the old index if possible so that we don't produce more requests for the first server
       if (index >= this.servers.length) {
-         index = 0;
+         index = ThreadLocalRandom.current().nextInt(this.servers.length);
```
Comment (tristantarrant) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/RoundRobinBalancingStrategy.java]: Shouldn't `index` be initialized to MAX_INT in order to trigger this on 1st call ?

Comment (danberindei) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/RoundRobinBalancingStrategy.java]: Oops, I meant to remove the `if`

**Comment by danberindei** [comment]
Updated to remove the if and to fix the test

**Comment by wburns** [comment]
Integrated into master, thanks @danberindei !

**Pull request merged**